int32 num_elements
float32[] scannerx
float32[] scannery
uint8[] scanner_types
